custom/icds_reports/filters.py

```diff
@@ -66,9 +78,11 @@ def loc_to_json(loc, project):
                 # there are some instances in viewing archived locations where we don't actually
                 # support drilling all the way down.
                 break
-            this_loc['children'] = [loc_to_json(loc, project) for loc in children]
```

Comment:
just spent roughly 30 mins trying to understand what is happening here, so going to put this down as suggestion to consider at least
my understanding is that this block just wants to find the children locations of all ancestors
```
selected_location = SQLLocation.active_objects.get(domain=domain, location_id=selected_loc_id)
json_at_level = loc_json  # json at level at which we should find the ancestor
for ancestor in selected_location.get_ancestors():  # this would start with top level ancestor first
    ancestor_loc_dict_in_json = None
    for loc_dict in json_at_level:
        if loc_dict['uuid'] == ancestor.location_id:
            ancestor_loc_dict_in_json = loc_dict
            break
    if not ancestor_loc_dict_in_json:  # could not find the ancestor at the level, user should not have reached at tihs point to try access an ancestor not permitted access to
        break
    if self.user.has_permission(domain, 'access_all_locations'):
        ancestor_loc_dict_in_json['children'] = [loc_to_json(loc, project) for loc in ancestor.child_locations()
    else:
        ancestor_loc_dict_in_json['children'] = [loc_to_json(loc, project) for loc in ancestor.child_locations()
                                                                      if loc.location_id in accessible_location_ids]
    json_at_level = ancestor_loc_dict_in_json['children']  # reset level to one level down to find next ancestor
```